### PR TITLE
don't call show() inside animation

### PR DESCRIPTION
**Pull request by pelikhan**
This is a change to the animation API so that animation don't call show on their own. The animation engine should be in charge of calling show.

This allows to run multiple animations at once and compose them.

### CODE REVIEW

**Comment by pelikhan** [comment]
@microsoftsam review?

File: libs/light/neopixel.ts [hunk 1]
```diff
@@ -568,7 +573,12 @@ namespace light {
         //% weight=87 blockGap=8
         showAnimationFrame(animation: NeoPixelAnimation) {
             if (!animation) return;
-            animation.showFrame(this);
+            const buf = this.buffered();
+            this.setBuffered(false);
+            animation.renderFrame(this);
+            this.setBuffered(buf);
+            this.show();
+            loops.pause(50);
```
Comment (microsoftsam) [inline libs/light/neopixel.ts]: No more variable delay between animation frames?

**Comment by pelikhan** [comment]
Actually showAnimationFrame should probably not have a delay in it. Each animation should handle timing internally and let the animation engine call them when needed.

File: libs/light/neopixel.ts (hunk 1 above)
Comment (pelikhan) [reply libs/light/neopixel.ts]: removed

**Pull request merged**